neuro_san/client/thinking_file_message_processor.py

```diff
@@ -89,31 +93,68 @@ def write_message(self, response: Dict[str, Any], origin_str: str):
             # There is no real text, but there is a structure. JSON-ify it.
             text = json.dumps(structure, indent=4, sort_keys=True)
 
+        # Figure out how we are going to report the origin given the message.
+        use_origin: str = self._determine_origin_reporting(response, origin_str)
```

Comment:
Break out the policy of how we are going to report the origin as separate method.